### PR TITLE
[CONTRIB] Improve Spark backend support for contrib query based expectations

### PR DESCRIPTION
**Pull request by mkopec87**
Changes proposed in this pull request:
- Enable Spark tests for 3 contrib query-based expectations
- Fix tests for two contrib query-based expectations

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code

### CODE REVIEW

File: contrib/experimental/great_expectations_experimental/expectations/expect_query_count_with_filter_to_meet_threshold.py [hunk 1]
```diff
@@ -24,12 +24,12 @@ class ExpectQueryCountWithFilterToMeetThreshold(QueryExpectation):
     metric_dependencies = ("query.template_values",)
 
     query = """
-                SELECT COUNT(*) n
+                SELECT COUNT(1)
```
Comment (kenwade4) [inline contrib/experimental/great_expectations_experimental/expectations/expect_query_count_with_filter_to_meet_threshold.py]: Can you change the SELECT to what it was before

File: contrib/experimental/great_expectations_experimental/expectations/expect_query_count_with_filter_to_meet_threshold.py [hunk 2]
```diff
@@ -75,7 +75,12 @@ def _validate(
 
     examples = [
         {
-            "data": {"col1": [1, 1, 1, 2, 2, 2, 2, 2]},
+            "data": [
+                {
+                    "dataset_name": "test",
```
Comment (kenwade4) [inline contrib/experimental/great_expectations_experimental/expectations/expect_query_count_with_filter_to_meet_threshold.py]: Remove this data_set_name key

File: contrib/experimental/great_expectations_experimental/expectations/expect_day_sum_to_be_close_to_equivalent_week_day_mean.py [hunk 3]
```diff
@@ -163,7 +163,8 @@ class ExpectDaySumToBeCloseToEquivalentWeekDayMean(QueryExpectation):
                 {
                     "backend": "sqlalchemy",
                     "dialects": ["sqlite"],
-                }
+                },
+                {"backend": "spark", "dialects": None},
```
Comment (kenwade4) [inline contrib/experimental/great_expectations_experimental/expectations/expect_day_sum_to_be_close_to_equivalent_week_day_mean.py:167]: Can you delete the whole `test_backends` section, delete the `only_for` line on line 103, and add `"suppress_test_for": ["bigquery"],` after line 88

```--- a/contrib/experimental/great_expectations_experimental/expectations/expect_day_sum_to_be_close_to_equivalent_week_day_mean.py
+++ b/contrib/experimental/great_expectations_experimental/expectations/expect_day_sum_to_be_close_to_equivalent_week_day_mean.py
@@ -86,6 +86,7 @@ class ExpectDaySumToBeCloseToEquivalentWeekDayMean(QueryExpectation):
                 "summed_column_zero_both": generate_data_sample({1: 3, 0: 12}),
             },
             # "column_b": [today, yesterday, yesterday, two_days_ago]},
+            "suppress_test_for": ["bigquery"],
             "tests": [
                 {
                     "title": "positive_test",
@@ -100,7 +101,6 @@ class ExpectDaySumToBeCloseToEquivalentWeekDayMean(QueryExpectation):
                         "threshold": default_kwarg_values["threshold"],
                     },
                     "out": {"success": True},
-                    "only_for": ["sqlite"],
                 },
                 {
                     "title": "negative test",
@@ -159,13 +159,6 @@ class ExpectDaySumToBeCloseToEquivalentWeekDayMean(QueryExpectation):
                     "out": {"success": False},
                 },
             ],
-            "test_backends": [
-                {
-                    "backend": "sqlalchemy",
-                    "dialects": ["sqlite"],
-                },
-                {"backend": "spark", "dialects": None},
-            ],
         }
     ]
```

File: contrib/experimental/great_expectations_experimental/expectations/expect_queried_column_list_to_be_unique.py [hunk 4]
```diff
@@ -121,9 +120,15 @@ def _validate(
                         }
                     },
                     "out": {"success": False},
-                    "only_for": ["sqlite"],
                 },
             ],
+            "test_backends": [
```
Comment (kenwade4) [inline contrib/experimental/great_expectations_experimental/expectations/expect_queried_column_list_to_be_unique.py]: Can you delete the whole `test_backends` section, then add `"suppress_test_for": ["postgresql", "bigquery"],` after line 100

```
--- a/contrib/experimental/great_expectations_experimental/expectations/expect_queried_column_list_to_be_unique.py
+++ b/contrib/experimental/great_expectations_experimental/expectations/expect_queried_column_list_to_be_unique.py
@@ -98,6 +98,7 @@ class ExpectQueriedColumnListToBeUnique(QueryExpectation):
                     },
                 },
             ],
+            "suppress_test_for": ["postgresql", "bigquery"],
             "tests": [
                 {
                     "title": "basic_positive_test",
@@ -122,13 +123,6 @@ class ExpectQueriedColumnListToBeUnique(QueryExpectation):
                     "out": {"success": False},
                 },
             ],
-            "test_backends": [
-                {
-                    "backend": "sqlalchemy",
-                    "dialects": ["sqlite"],
-                },
-                {"backend": "spark", "dialects": None},
-            ],
         }
     ]
```

File: contrib/experimental/great_expectations_experimental/expectations/expect_queried_column_value_frequency_to_meet_threshold.py [hunk 5]
```diff
@@ -178,7 +178,7 @@ def _validate(
                     "in": {
                         "column": "col2",
                         "value": ["a", "b"],
-                        "threshold": [0.6, 0.4],
+                        "threshold": [0.5, 0.5],
```
Comment (kenwade4) [inline contrib/experimental/great_expectations_experimental/expectations/expect_queried_column_value_frequency_to_meet_threshold.py:181]: On line 172, can you change sqlite to sqlalchemy, then add `"suppress_test_for": ["bigquery", "trino"],` after line 118, then delete the other 4 only_for lines

```
--- a/contrib/experimental/great_expectations_experimental/expectations/expect_queried_column_value_frequency_to_meet_threshold.py
+++ b/contrib/experimental/great_expectations_experimental/expectations/expect_queried_column_value_frequency_to_meet_threshold.py
@@ -116,6 +116,7 @@ class ExpectQueriedColumnValueFrequencyToMeetThreshold(QueryExpectation):
                     },
                 },
             ],
+            "suppress_test_for": ["bigquery", "trino"],
             "tests": [
                 {
                     "title": "basic_positive_test",
@@ -127,7 +128,6 @@ class ExpectQueriedColumnValueFrequencyToMeetThreshold(QueryExpectation):
                         "threshold": 0.6,
                     },
                     "out": {"success": True},
-                    "only_for": ["sqlite", "spark"],
                 },
                 {
                     "title": "basic_negative_test",
@@ -139,7 +139,6 @@ class ExpectQueriedColumnValueFrequencyToMeetThreshold(QueryExpectation):
                         "threshold": 1,
                     },
                     "out": {"success": False},
-                    "only_for": ["sqlite", "spark"],
                 },
                 {
                     "title": "multi_value_positive_test",
@@ -151,7 +150,6 @@ class ExpectQueriedColumnValueFrequencyToMeetThreshold(QueryExpectation):
                         "threshold": [0.6, 0.4],
                     },
                     "out": {"success": True},
-                    "only_for": ["sqlite", "spark"],
                 },
                 {
                     "title": "multi_value_positive_test_static_data_asset",
@@ -169,7 +167,7 @@ class ExpectQueriedColumnValueFrequencyToMeetThreshold(QueryExpectation):
                                  """,
                     },
                     "out": {"success": True},
-                    "only_for": ["sqlite"],
+                    "only_for": ["sqlalchemy"],
                 },
                 {
                     "title": "multi_value_positive_test_row_condition",
@@ -183,7 +181,6 @@ class ExpectQueriedColumnValueFrequencyToMeetThreshold(QueryExpectation):
                         "condition_parser": "great_expectations__experimental__",
                     },
                     "out": {"success": True},
-                    "only_for": ["sqlite", "spark"],
                 },
             ],
         },
```

File: contrib/experimental/great_expectations_experimental/expectations/expect_queried_slowly_changing_table_to_have_no_gaps.py [hunk 6]
```diff
@@ -205,14 +204,14 @@ def _validate(
...
                     "out": {"success": False},
-                    "only_for": ["sqlite"],
                 },
             ],
             "test_backends": [
                 {
                     "backend": "sqlalchemy",
                     "dialects": ["sqlite"],
-                }
+                },
```
Comment (kenwade4) [inline contrib/experimental/great_expectations_experimental/expectations/expect_queried_slowly_changing_table_to_have_no_gaps.py:213]: Can you delete the whole `test_backends` section, then add `"suppress_test_for": ["postgresql", "bigquery"],` after line 179

```
--- a/contrib/experimental/great_expectations_experimental/expectations/expect_queried_slowly_changing_table_to_have_no_gaps.py
+++ b/contrib/experimental/great_expectations_experimental/expectations/expect_queried_slowly_changing_table_to_have_no_gaps.py
@@ -177,6 +177,7 @@ class ExpectQueriedSlowlyChangingTableToHaveNoGaps(QueryExpectation):
                     },
                 },
             ],
+            "suppress_test_for": ["postgresql", "bigquery"],
             "tests": [
                 {
                     "title": "basic_positive_test",
@@ -206,13 +207,6 @@ class ExpectQueriedSlowlyChangingTableToHaveNoGaps(QueryExpectation):
                     "out": {"success": False},
                 },
             ],
-            "test_backends": [
-                {
-                    "backend": "sqlalchemy",
-                    "dialects": ["sqlite"],
-                },
-                {"backend": "spark", "dialects": None},
-            ],
         },
     ]
```

File: contrib/experimental/great_expectations_experimental/expectations/expect_query_count_with_filter_to_meet_threshold.py [hunk 7]
```diff
@@ -75,7 +75,12 @@ def _validate(
...
+                    "data": {"col1": [1, 1, 1, 2, 2, 2, 2, 2]},
+                },
+            ],
```
Comment (kenwade4) [inline contrib/experimental/great_expectations_experimental/expectations/expect_query_count_with_filter_to_meet_threshold.py:82]: Can you add `"suppress_test_for": ["bigquery"],` before the `"tests": [` line, then delete the two only_for lines

```
--- a/contrib/experimental/great_expectations_experimental/expectations/expect_query_count_with_filter_to_meet_threshold.py
+++ b/contrib/experimental/great_expectations_experimental/expectations/expect_query_count_with_filter_to_meet_threshold.py
@@ -24,7 +24,7 @@ class ExpectQueryCountWithFilterToMeetThreshold(QueryExpectation):
     metric_dependencies = ("query.template_values",)

     query = """
-                SELECT COUNT(1)
+                SELECT COUNT(*) n
                 FROM {active_batch}
                 WHERE {col} = {filter}
             """
@@ -77,10 +77,10 @@ class ExpectQueryCountWithFilterToMeetThreshold(QueryExpectation):
         {
             "data": [
                 {
-                    "dataset_name": "test",
                     "data": {"col1": [1, 1, 1, 2, 2, 2, 2, 2]},
                 },
             ],
+            "suppress_test_for": ["bigquery"],
             "tests": [
                 {
                     "title": "basic_positive_test",
@@ -91,7 +91,6 @@ class ExpectQueryCountWithFilterToMeetThreshold(QueryExpectation):
                         "threshold": 4,
                     },
                     "out": {"success": True},
-                    "only_for": ["sqlite", "spark"],
                 },
                 {
                     "title": "basic_negative_test",
@@ -102,7 +101,6 @@ class ExpectQueryCountWithFilterToMeetThreshold(QueryExpectation):
                         "threshold": 4,
                     },
                     "out": {"success": False},
-                    "only_for": ["sqlite", "spark"],
                 },
             ],
         },
```

**Review by kenwade4 — CHANGES_REQUESTED** [review]
Thanks for digging into these query based Expectations @mkopec87. There are a bunch of changes requested, but it might be easier for you to **just make two of the requested changes on expect_query_count_with_filter_to_meet_threshold.py (reverting the SELECT and deleting dataset_name key from data in examples)**, then I'll add those other changes in my own new PR after merging yours.

**Comment by mkopec87** [comment]
@kenwade4 I made just the two changes you mentioned + resolved conflicts with develop. Let me know if it works now :)

**Review by kenwade4 — APPROVED** [review]

**Pull request merged**